Python/Product/PythonTools/PythonTools/Project/PythonProjectNode.cs

```diff
@@ -204,6 +202,16 @@ internal partial class PythonProjectNode :
             Site.GetUIThread().Invoke(() => RefreshInterpreters());
         }
 
+        // Called once all async interpreter factories have finished discovering interpreters
+        private void OnInterpreterDiscoveryCompleted(object sender, EventArgs e) {
+            if (!_asyncInfoBarCheckTriggered) {
+                _asyncInfoBarCheckTriggered = true;
+
+                // Check for any missing environments and show info bars for them
+                _condaEnvCreateInfoBar.CheckAsync().HandleAllExceptions(Site, typeof(PythonProjectNode)).DoNotWait();
```

Comment:
what about open folder mode?

should be just recheck all infobars? there is a conda one too `_condaEnvCreateInfoBar`

Reply:
I checked open folder mode, but there is no PythonProject node in that case. We don't have a Python Environments node, we just use the normal environment list that's in VS and you can select the one you want from the dropdown when you are going to run the program. Since these bugs manifest in the project node, I don't believe these bugs repro when opening in folder view. At least, I couldn't repro them. 😄

Reply:
Ok. I thought because it was also calling _condaEnvCreateInfoBar in open folder that it should refresh on conda environments detected